src/NuGet.Clients/NuGet.CommandLine/Common/MSBuildProjectSystem.cs

```diff
@@ -35,7 +35,7 @@ public class MSBuildProjectSystem : MSBuildUser, IMSBuildNuGetProjectSystem
             NuGetProjectContext = projectContext;
         }
 
-        public INuGetProjectContext NuGetProjectContext { get; private set; }
+        public INuGetProjectContext NuGetProjectContext { get; set; }
```

Comment:
This makes sense. Although existing code didn't enforce null-check for this property. I'd avoid changing this behavior in the scope of this PR.